asab/storage/elasticsearch.py

```diff
@@ -195,34 +318,41 @@ async def put_index_template(self, template_name: str, template: dict) -> dict:
 			:raise Exception: Raised if connection to all server URLs fails.
 		"""
 		for url in self.ServerUrls:
-			url = "{}_template/{}?include_type_name".format(url, template_name)
-			L.warning("sending into url: {}".format(url))
+			request_url = "{}_template/{}?include_type_name".format(url, template_name)
+			L.warning("Posting index template into url: {}".format(request_url))
 			try:
-				async with self.session().request(method="POST", url=url, data=json.dumps(template), headers={
-					'Content-Type': 'application/json'
-				}) as resp:
+				async with self.session().request(
+					method="POST",
+					url=request_url,
+					data=json.dumps(template),
+					headers=self.Headers,
+					ssl=self.SSLcontext,
+				) as resp:
 					assert resp.status == 200, "Unexpected response code: {}".format(resp.status)
 					resp = await resp.json()
+					await self.session().close()
 					return resp
 			except aiohttp.client_exceptions.ClientConnectorError:
-				if url == self.Storage.ServerUrls[-1]:
+				if url == self.ServerUrls[-1]:
 					raise Exception("Failed to connect to '{}'".format(url))
 				else:
 					L.warning("Failed to connect to '{}', iterating to another cluster node".format(url))
+					return {}
 
 
 	async def reindex(self, previous_index, new_index):
 		for url in self.ServerUrls:
 			try:
 				if url.endswith('/'):
-					url = "{}_reindex".format(url)
+					request_url = "{}_reindex".format(url)
 				else:
-					url = "{}/_reindex".format(url)
+					request_url = "{}/_reindex".format(url)
 
```

Comment:
If urls from configs are parsed, checked and corrected properly (namely "/" is added at the end), we don't need this condition. If not - don't we need to have it at every endpoint?